modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -100,20 +96,20 @@
         intSetting("transport.netty.boss_count", 1, 1, Property.NodeScope);
 
 
+    private final SharedGroupFactory sharedGroupFactory;
     private final RecvByteBufAllocator recvByteBufAllocator;
-    private final int workerCount;
     private final ByteSizeValue receivePredictorMin;
     private final ByteSizeValue receivePredictorMax;
     private final Map<String, ServerBootstrap> serverBootstraps = newConcurrentMap();
     private volatile Bootstrap clientBootstrap;
-    private volatile NioEventLoopGroup eventLoopGroup;
+    private volatile SharedGroupFactory.SharedGroup sharedGroup;
```

Comment:
do we need to store this here? Should we just get it from `sharedGroupFactory.get...`?